### PR TITLE
Fix sidecars not retrieving updated mesh networks configuration

### LINKED ISSUE #11221: Mesh networks config modifications don't propagate
Having a setup of Multi-cluster with Mesh networks configured in the Istio config map.
Modify the address/port of the network gateway within the deployed config map and examine updated pushed EDS configuration in one of sidecars.

While Pilot's log shows that the changes were detected and configmap reloaded, the initial network gateway values are still being pushed while expecting the new values to take affect.

Workaround: Kill the pilot pod after modifying the configmap.

### PR DESCRIPTION
**Pull request by ymesika**
Fixes #11221 

/assign @hzxuzhonghu

### CODE REVIEW

File: pilot/pkg/bootstrap/server.go [hunk 1]
```diff
@@ -455,6 +455,7 @@ func (s *Server) initMeshNetworks(args *PilotArgs) error {
 			log.Infof("mesh networks configuration file updated to: %s", spew.Sdump(meshNetworks))
 			util.ResolveHostsInNetworksConfig(s.meshNetworks)
 			s.meshNetworks = meshNetworks
+			s.EnvoyXdsServer.Env.MeshNetworks = meshNetworks
```
Comment (hzxuzhonghu) [inline pilot/pkg/bootstrap/server.go]: Also `InitNetworkLookup` should be called again.

File: pilot/pkg/bootstrap/server.go [hunk 2]
```diff
@@ -455,6 +455,7 @@ func (s *Server) initMeshNetworks(args *PilotArgs) error {
...
 			s.EnvoyXdsServer.ConfigUpdate(true)
```
[the remark this answers is not in the document]
Comment (ymesika) [reply pilot/pkg/bootstrap/server.go]: This piece of code is for handling configmap file change events so theoretically the server is already up and running. But I will check anyway.

File: pilot/pkg/bootstrap/server.go [hunk 3]
```diff
@@ -455,7 +457,11 @@ func (s *Server) initMeshNetworks(args *PilotArgs) error {
 			log.Infof("mesh networks configuration file updated to: %s", spew.Sdump(meshNetworks))
 			util.ResolveHostsInNetworksConfig(s.meshNetworks)
 			s.meshNetworks = meshNetworks
-			s.EnvoyXdsServer.ConfigUpdate(true)
+			s.kubeRegistry.InitNetworkLookup(meshNetworks)
```
Comment (hzxuzhonghu) [inline pilot/pkg/bootstrap/server.go]: `s.kubeRegistry` also need protect

File: pilot/pkg/bootstrap/server.go [hunk 4]
```diff
@@ -391,7 +391,9 @@ func (s *Server) initMesh(args *PilotArgs) error {
 					//TODO Need to re-create or reload initConfigController()
 				}
 				s.mesh = mesh
-				s.EnvoyXdsServer.ConfigUpdate(true)
+				if s.EnvoyXdsServer != nil {
+					s.EnvoyXdsServer.ConfigUpdate(true)
```
Comment (hzxuzhonghu) [inline pilot/pkg/bootstrap/server.go]: add ` s.EnvoyXdsServer.Env.Mesh = mesh` to fix mesh configuration reload

**Pull request merged**